Sources/CodeEditTextView/TreeSitter/TreeSitterClient+Edit.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  TreeSitterClient+Edit.swift
+//  
+//
+//  Created by Khan Winter on 3/10/23.
+//
+
+import Foundation
+import SwiftTreeSitter
+import CodeEditLanguages
+
+extension TreeSitterClient {
+
+    /// Calculates a series of ranges that have been invalidated by a given edit.
+    /// - Parameters:
+    ///   - textView: The text view to use for text.
+    ///   - edit: The edit to act on.
+    ///   - language: The language to use.
+    ///   - readBlock: A callback for fetching blocks of text.
+    /// - Returns: An array of distinct `NSRanges` that need to be re-highlighted.
+    func findChangedByteRanges(textView: HighlighterTextView,
+                               edit: InputEdit,
+                               layer: LanguageLayer,
+                               readBlock: @escaping Parser.ReadBlock) -> [NSRange] {
```

Comment:
### [CodeStyle]:

Maybe try to not indent to much. In the `CodeEdit` repo this is not allowed by SwiftLint. We should keep it consistent.

```suggestion
    func findChangedByteRanges(
        textView: HighlighterTextView,
        edit: InputEdit,
        layer: LanguageLayer,
        readBlock: @escaping Parser.ReadBlock
    ) -> [NSRange] {
```